pkg/mcs/resource_manager/server/server.go

```diff
@@ -403,14 +399,14 @@ func (s *Server) startServer() (err error) {
 	}
 
 	// Server has started.
-	atomic.StoreInt64(&s.isServing, 1)
 	entry := &discovery.ServiceRegistryEntry{ServiceAddr: s.cfg.ListenAddr}
 	serializedEntry, err := entry.Serialize()
 	if err != nil {
 		return err
 	}
 	s.serviceRegister = discovery.NewServiceRegister(s.ctx, s.etcdClient, utils.ResourceManagerServiceName, s.cfg.ListenAddr, serializedEntry, discovery.DefaultLeaseInSeconds)
 	s.serviceRegister.Register()
+	atomic.StoreInt64(&s.isServing, 1)
```

Comment:
why move it

Reply:
We should move it, shouldn't we? After we register the service, then change the state to serving, will be more reasonable.

Reply:
By the way, the order doesn't impact any running result. It seems to me that, starting the service, registering the service then changing the state to 'isServing' seems to be more reasonable.